### PR TITLE
auto derive myriadinput decoder

### PR DESCRIPTION
**Pull request by danbills**

### CODE REVIEW

File: cwl/src/main/scala-2.11/wdl4s/cwl/CwlInputParsing.scala [hunk 1]
```diff
@@ -13,18 +14,15 @@ import lenthall.validation.Checked._
...
-  private [cwl] lazy val inputCoercionFunction: InputParsingFunction = inputFile => {
-    yaml.parser.parse(inputFile).flatMap(_.as[Map[String, MyriadInputValue]]) match {
+  implicit val x = implicitly[Decoder[File]]
+
+  private [cwl] lazy val inputCoercionFunction: InputParsingFunction =
+    inputFile => {
+    yaml.
+      parser.
+      parse(inputFile).
+      flatMap(json => decode[Map[String, MyriadInputValue]](io.circe.Printer.noSpaces.pretty(json))) match {
```
Comment (Horneth) [inline cwl/src/main/scala-2.11/wdl4s/cwl/CwlInputParsing.scala]: do we really have to parse the yaml, write it back to json, and reparse it again ? :/

Comment (danbills) [reply cwl/src/main/scala-2.11/wdl4s/cwl/CwlInputParsing.scala]: No this is the naive way I was doing originally and in fact your way is correct.  changing this

File: cwl/src/main/scala-2.12/wdl4s/cwl/CwlInputParsing.scala [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-  implicit val fileD = implicitly[Decoder[File]]
-  implicit val stringD = implicitly[Decoder[String]]
-  implicit val intD = implicitly[Decoder[Int]]
-  implicit val longD = implicitly[Decoder[Long]]
-  implicit val floatD = implicitly[Decoder[Float]]
-  implicit val doubleD = implicitly[Decoder[Double]]
-  implicit val booleanD = implicitly[Decoder[Boolean]]
-  implicit val inputArray = implicitly[Decoder[Array[MyriadInputValuePrimitives]]]
+
+  implicit val f = implicitly[Decoder[File]]
```
Comment (danbills) [inline cwl/src/main/scala-2.12/wdl4s/cwl/CwlInputParsing.scala]: there is some issue with deriving decoders that makes this necessary.  I haven't quite pinpointed it but it seems to involve a member of a `Coproduct` having a Witness field.

**Comment by mcovarr** [comment]
I dig the "less code" part, is the compilation time improved as well?

**Comment by danbills** [comment]
@mcovarr I don't have numbers but it's slightly improved as it's deriving decoders fewer times.

**Comment by Horneth** [comment]
Rebased version here https://github.com/broadinstitute/wdl4s/pull/241

**Pull request closed without merging**